fbw-common/src/systems/shared/src/GsxSync.ts

```diff
@@ -0,0 +1,407 @@
+
+      //need an Fuel Increase later, so target is lowered for desired >= FOB Cases
+      if (this.isDefuel) {
+        if (this.desiredFuel >= this.DEFUEL_DIFF_TARGET) {
+          this.setDesiredFuel(this.desiredFuel - this.DEFUEL_DIFF_TARGET);
+        } else {
+          this.setDesiredFuel(0);
+        }
+      }
+      this.isRefuelActive = true;
+      this.fuelTimer.schedule(() => {
+        this.setRefuelStartedState(true);
+      }, DELAY_FUEL_START);
+    }
+
+    //something bad happend, hose disconnected while still refueling - cleanup
+    if (hose === GsxStates.DISCONNECTED && (this.refuelStartedByUser.get() || this.isRefuelActive)) {
+      if (this.isDefuel) {
+        this.setDesiredFuel(this.desiredFuel);
+      }
+
+      this.isDefuel = false;
+      this.desiredFuel = 0;
+      this.isRefuelActive = false;
+    }
+  }
+
+  protected onRefuelStartedByUser(refuel: boolean): void {
+    if (!this.isSyncEnabled(SyncServices.FUEL)) {
+      return;
+    }
+
+    //defuel has finished - instant refuel to desired to get the truck triggered to leave
+    if (!refuel && this.fuelHose.get() === GsxStates.CONNECTED && this.isDefuel) {
+      this.setDesiredFuel(this.desiredFuel);
+      const refuelRate = this.getRefuelRate();
+      this.setRefuelRate(RefuelRate.INSTANT);
+
+      this.fuelTimer.schedule(() => {
+        this.setRefuelStartedState(true);
+        this.isDefuel = false;
+        this.desiredFuel = 0;
+      }, DELAY_REFUEL_RESTART);
+
+      this.rateTimer.schedule(() => {
+        this.setRefuelRate(refuelRate);
+      }, DELAY_RATE_RESTORE);
+    }
+
+    if (!refuel) {
+      this.isRefuelActive = false;
+    }
+  }
+
+  protected evaluateGsxPowerSource(): void {
+    if (!this.isSyncEnabled(SyncServices.POWER)) {
+      return;
+    }
+
+    const extAvail = this.isExtPowerAvail();
+    const jetwayState = this.stateJetway.get();
+    const gpuState = this.stateGpu.get();
+    const boardState = this.stateBoard.get();
+    const departureState = this.stateDeparture.get();
+    let action = PowerAction.NOOP;
+
+    //external is available (as of internal State) - check if disconnect is needed
+    if (extAvail) {
+      //pushback was requested (when boarding not running)
+      if (
+        boardState !== GsxServiceStates.REQUESTED &&
+        boardState !== GsxServiceStates.ACTIVE &&
+        departureState >= GsxServiceStates.REQUESTED
+      ) {
+        action = PowerAction.DISCONNECT;
+        //pushback is active (in it wasn't catched before)
+      } else if (departureState === GsxServiceStates.ACTIVE) {
+        action = PowerAction.DISCONNECT;
+        //jetway and gpu both not connected
+      } else if (
+        jetwayState >= GsxServiceStates.CALLABLE &&
+        jetwayState <= GsxServiceStates.REQUESTED &&
+        gpuState !== GsxServiceStates.ACTIVE
+      ) {
+        action = PowerAction.DISCONNECT;
+      }
+      //external is not available - check if connect is needed
```

Comment:
Always a space for readability.
```suggestion
    // external is available (as of internal State) - check if disconnect is needed
    if (extAvail) {
      // pushback was requested (when boarding not running)
      if (
        boardState !== GsxServiceStates.REQUESTED &&
        boardState !== GsxServiceStates.ACTIVE &&
        departureState >= GsxServiceStates.REQUESTED
      ) {
        action = PowerAction.DISCONNECT;
        // pushback is active (in it wasn't catched before)
      } else if (departureState === GsxServiceStates.ACTIVE) {
        action = PowerAction.DISCONNECT;
        // jetway and gpu both not connected
      } else if (
        jetwayState >= GsxServiceStates.CALLABLE &&
        jetwayState <= GsxServiceStates.REQUESTED &&
        gpuState !== GsxServiceStates.ACTIVE
      ) {
        action = PowerAction.DISCONNECT;
      }
      // external is not available - check if connect is needed
```

Reply:
Yeah good Idea, most of my usual Comments have zero Length 😂